crates/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -530,6 +539,10 @@ impl Context {
         last_array_uses: &HashMap<ValueId, InstructionId>,
     ) -> Result<(), RuntimeError> {
         let index_const = dfg.get_numeric_constant(index);
+        // we use predicate*index instead of index if there are side effects, in order to avoid index-out-of-bound with false predicates.
```

Comment:
```suggestion
        // We use predicate*index instead of index if there are side effects, in order to avoid index-out-of-bound with false predicates.
```